tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -1392,21 +1392,29 @@ std::optional<Value> convertExpandDimsOp(PatternRewriter& rewriter,
   }
   int32_t dim = dim_elem.getValues<IntegerAttr>()[0].getInt();
   int32_t input_size = input_shape.size();
+
+  // check: -1-input_size <= dim <= input_size
+  if (dim < (-1 - input_size)) {
+    (void)rewriter.notifyMatchFailure(
+        op, "dimension to expand cannot be less than -1-input.dims().");
```

Comment:
nit: do we really have a API like `input.dims()`? this function seems like returning all the dims instead of returning the size of dims(), or simply the rank.